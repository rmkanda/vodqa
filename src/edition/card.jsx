import React, { Component } from "react";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";

import "./index.scss";

export class EditionCard extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    const edition = this.props.edition;
    const type = edition.type || "TALK";
    const title = edition.title || `EDITION ${edition.id} `;
    const startTime = new Date(edition.startTime);
    const endTime = new Date(edition.endTime);
    const formattedDate = startTime.toLocaleDateString("en-IN", {
      day: "numeric",
      month: "short",
      year: "numeric",
    });
    const duration = Math.abs(startTime - endTime) / 36e5;
    const sessions = edition?.sessions || [];
    const sessionCount = sessions?.length || 0;
    const getTalksCount = (sessions) => {
      let count = 0;
      sessions.map((session) => (count += session.talks.length));
      return count;
    };
    const talksCount = getTalksCount(sessions) || 0;
    const bannerId = isNaN(edition.id % 4) ? 0 : edition.id % 4;

    return (
      <Link
        key={edition.id}
        className={`editionCard event-bg-${bannerId}`}
        to={`./edition?editionId=${edition.id}`}
      >
        <span className="type">{type.toUpperCase()}</span>
        <div className="details">
          <p className="title">{title}</p>
          <p className="date">
            Edition {edition.id} | {formattedDate}
          </p>
          <div className="info">
            <div className="item">
              <p className="value">{String(talksCount).padStart(2, "0")}</p>
              <p>{talksCount > 1 ? "Talks" : "Talk"}</p>
            </div>
            <div className="item">
              <p className="separator" />
              <p className="value">{String(sessionCount).padStart(2, "0")}</p>
              <p>{sessionCount > 1 ? "Sessions" : "Session"}</p>
            </div>
            <div className="item">
              <p className="separator" />
              <p className="value">{String(duration).padStart(2, "0")} hr</p>
              <p>Duration</p>
            </div>
          </div>
        </div>
      </Link>
    );
  }
}

EditionCard.propTypes = {
  edition: PropTypes.object.isRequired,
};
