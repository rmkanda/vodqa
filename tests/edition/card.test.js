import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import { EditionCard } from "../../src/edition/card.jsx";

describe("Cards Section", () => {
  test("edition card with mising details", () => {
    const { asFragment } = render(<EditionCard edition={{}} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
