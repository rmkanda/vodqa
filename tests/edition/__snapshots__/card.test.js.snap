// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards Section edition card with mising details 1`] = `
<DocumentFragment>
  <div
    class="editionCard event-bg"
  >
    <span
      class="type"
    >
      TALK
    </span>
    <div
      class="details"
    >
      <p
        class="title"
      >
        EDITION undefined 
      </p>
      <p
        class="date"
      >
        Edition  | Invalid Date
      </p>
      <div
        class="info"
      >
        <div
          class="item"
        >
          <p
            class="value"
          >
            00
          </p>
          <p>
            Talk
          </p>
        </div>
        <div
          class="item"
        >
          <p
            class="separator"
          />
          <p
            class="value"
          >
            00
          </p>
          <p>
            Session
          </p>
        </div>
        <div
          class="item"
        >
          <p
            class="separator"
          />
          <p
            class="value"
          >
            NaN hr
          </p>
          <p>
            Duration
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
